tephra-core/src/main/java/co/cask/tephra/visibility/WriteFence.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.tephra.visibility;
+
+import co.cask.tephra.Transaction;
+import co.cask.tephra.TransactionAware;
+import com.google.common.primitives.Bytes;
+import com.google.common.primitives.Longs;
+import com.google.common.primitives.UnsignedBytes;
+
+import java.util.Collection;
+import java.util.TreeSet;
+
+/**
+ * Implementation to wait for a {@link VisibilityFence}.
```

Comment:
This doc is not very accurate as the waiting doesn't happen in this class, but rather in the FenceWait class.